RELEASE_NOTE_DRAFT.md

```diff
@@ -1,101 +1,86 @@
+* Added peekAndPop methods in ImmutableStack.
+* Added fused method for collect + makeString.
+* Added missing implementations of toString().
+* Added selectWithIndex and rejectWithIndex to Ordered Primitive Iterables.
+* Implemented containsAny, containsNone, containsAnyIterable, containsNoneIterable on RichIterable.
 
 # Optimizations
 ----------------------
-* Optimized ImmutableListFactoryImpl.withAll(Iterable).
-* Optimized containsAll on primitive iterables.
-* Optimized primitive hash set newSet and withAll.
-* Optimized sumByLong and sumByInt Primitive methods for Bags.
-* Optimized toImmutable on primitive sets.
-* Optimized union and difference on SetIterables.
-* Added enhanced for-loop syntax wherever as possible.
+* Optimized asParallel() for immutable sets in O(1) instead of O(n).
+* Optimized some implementations of toString() that were delegating to iterators.
+* Improved equals() performance for Set implementations.
 
 # Tech Debt Reduction
 ---------------------
-* Deprecated  ImmutableArrayStack.
-* Fixed bug with noneSatisfy for ImmutableBooleanEmptySet.
-* Fixed compiler errors for JDK-15-EA.
-* Fixed return types of aggregateBy().
-* Fixed types on Multimap.*MultiValues().
-* Fixed primitive hash map values collections removed  for special key zero.
-* Fixed CollectIterable detect methods.
-* Added missing overrides for toImmutable methods on synchronized primitive object maps.
-* Added missing overrides of aggregateBy() and aggregateInPlaceBy().
-* Added documentation on serializing Eclipse Collections with jackson.
-* Added logic to throw UnsupportedOperationException on calling withMap method in FixedSizeMap. This is a breaking change.
-* Made org.eclipse.collections.impl.list.Interval#goForward private. This is a breaking change.
-* Replaced HashingStrategies.longHashCode with Java 8 Long.hashCode. This is a breaking change.
-* Deleted version.properties as the file is no longer needed.
-* Removed extra calls to map.get in sumByDoubleFunction and sumByFloatFunction.
-* Removed implementations of OrderedIterable.toStack() that can use the default implementation instead.
-* Removed implementations of aggregateBy and aggregateInPlaceBy that can use default implementations instead.
-* Removed references to deleted classes in findbugs-exclude file.
+* Replaced implementation factories and dependencies with API factories where possible.
+* Fixed CheckStyle configuration for NewlineAtEndOfFile so that it works across operating systems.
+* Made forEach a default method on primitiveIterable.stg.
+* Made noneSatisfy a default method on primitiveIterable.stg.
+* Removed unnecessary implementations of toSortedList/Set/Bag/MapBy.
+* Improved code generation logic into separate goals for sources, test-sources, and resources.
+* Improved PIT mutation test coverage.
+* Improved overall test coverage by adding missing tests. 
 
 # Removed Functionality
```

Comment:
This section should be renamed to Deprecated Functionality or can be collapsed into Tech Debt Reduction.